modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/DeployerToolVersionsFetcher.java

```diff
@@ -0,0 +1,94 @@
+import java.util.List;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.models.servicetemplate.enums.DeployerKind;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Beans for fetching all available versions of the deployer tool.
+ */
+@Slf4j
+@Component
+public class DeployerToolVersionsFetcher {
+
+    private static final String OPEN_TOFU_VERSIONS_GITHUB_URL =
+            "https://github.com/opentofu/opentofu/releases";
+    private static final String TERRAFORM_VERSIONS_GITHUB_URL =
+            "https://github.com/hashicorp/terraform/releases";
+
+    private static final Pattern OFFICIAL_VERSION_PATTERN =
+            Pattern.compile("^v(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})$");
+
+
+    /**
+     * Fetch all official versions of the deployer tool.
+     *
+     * @param deployerKind deployer kind.
+     * @return list of all official versions.
+     */
+    @Cacheable(cacheNames = DEPLOYER_VERSIONS_CACHE_NAME)
+    public List<String> fetchOfficialVersionsOfDeployerTool(DeployerKind deployerKind) {
+        List<String> allVersions = new ArrayList<>();
+        int currentPage = 1;
+        String apiUrl = getFetchVersionsWebsite(deployerKind);
+        boolean hasNextPage = true;
+        while (hasNextPage) {
+            try {
+                String releaseUrlWithPage = apiUrl + "?page=" + currentPage;
+                List<String> versionsInCurrentPage = fetchVersionsFromWebsite(releaseUrlWithPage);
+                if (CollectionUtils.isEmpty(versionsInCurrentPage)) {
+                    hasNextPage = false;
+                } else {
+                    allVersions.addAll(versionsInCurrentPage);
+                    currentPage++;
+                }
+            } catch (IOException e) {
+                String errorMsg = String.format("Failed to fetch versions of the deployer tool %s "
+                        + "from url %s", deployerKind, apiUrl);
+                log.error(errorMsg, e);
+            }
+        }
+        return allVersions;
+    }
+
+    private String getFetchVersionsWebsite(DeployerKind deployerKind) {
+        return switch (deployerKind) {
+            case TERRAFORM -> TERRAFORM_VERSIONS_GITHUB_URL;
+            case OPEN_TOFU -> OPEN_TOFU_VERSIONS_GITHUB_URL;
+        };
+    }
+
+    private List<String> fetchVersionsFromWebsite(String releasesUrlWithPage) throws IOException {
+        List<String> versions = new ArrayList<>();
+        Document doc = Jsoup.connect(releasesUrlWithPage)
+                .ignoreHttpErrors(true).timeout(5000).get();
+        // version elements with css like <span class="ml-1 wb-break-all">v1.8.0</span>
+        Elements versionElements = doc.select("span.ml-1.wb-break-all");
```

Comment:
@baixinsui this looks a bit hacky. We did not use such solution in terraform-boot know?

How about using 'https://api.github.com/repos/hashicorp/terraform/releases' instead? May be we can even use the github Java SDK - https://github-api.kohsuke.org/. 
The releases and tags API of Github works without authentication for public repositories. 
Just an idea.

Reply:
I used request the API without the authentication. Due to the limit of 60 times per hour, some tests of the APIs failed.